kong/plugins/zipkin/tracing_headers.lua

```diff
@@ -314,6 +315,41 @@ local function parse_jaeger_trace_context_headers(jaeger_header)
 end
 
 
+local function parse_datadog_headers(headers)
+  -- allow testing to spy on this.
+  local warn = kong.log.warn
+
+  local should_sample = headers["x-datadog-sampling-priority"]
+  if should_sample == "1" or should_sample == "2" then 
+    should_sample = true
+  elseif should_sample == "0" or should_sample == "-1" then
+    should_sample = false
+  elseif should_sample ~= nil then
+    warn("x-datadog-sampling-priority header invalid; ignoring.")
```

Comment:
Wouldn't it make more sense to not register this trace if the sampling priority header was incorrect?
```suggestion
    warn("x-datadog-sampling-priority header invalid; ignoring.")
    return nil, nil, nil
```

Reply:
It was a mistake, thanks!